Chart editor - Mirror notes command

Lasercar opened the pull request:
<!-- Please read the Contributing Guide (https://github.com/FunkinCrew/Funkin/blob/main/docs/CONTRIBUTING.md) before submitting this PR. -->
## Does this PR close any issues? If so, link them below.
Closes #4365
## Briefly describe the issue(s) fixed.
Adds a command and songDataUtil to flip a selection of notes. Defaults to mirroring the notes on each strumline in the selection individually. Includes an option to flip the notes if the selection/notes passed into the function are in both strumlines, so that the resulting mirrored notes aren't inverted.

CTRL + SHIFT + M = Mirror Notes along X (note direction) axis

CTRL + ALT + M = Mirror Notes along the Y (note time) axis

## Include any relevant screenshots or videos.

https://github.com/user-attachments/assets/6e524957-1027-46c2-92d4-8b5b865b9400

Lasercar commented:
Hmm, maybe I should actually allow it to mirror X and Y, so that note kinds can be mirrored without the pattern of the notes changing?

MAZ12211 commented:
Can you also make it an option in the Edit thingy? and a way to mirror the notes vertically or horizontal or both?
Pretty much similar how SMEditor does it here
https://github.com/FunkinCrew/Funkin/issues/4365

Hundrec commented:
Yep, the submenu should go right after the flip notes menu.
I think it should be buttons rather than toggles.

![image](https://github.com/user-attachments/assets/87cc4af2-5048-437c-825b-468269a9d0d7)

Lasercar commented:
> > Can you also make it an option in the Edit thingy? and a way to mirror the notes vertically or horizontal or both?
Pretty much similar how SMEditor does it here
https://github.com/FunkinCrew/Funkin/issues/4365
>
> Yep, the submenu should go right after the flip notes menu. I think it should be buttons rather than toggles.

Yeah, I'll do that after the improved context menus get merged, so that I don't create a conflict when I make a branch with the mirror options added as buttons to the context menus and the edit tab.

Lasercar commented:
Ok, I fixed the merge conflict on this.

Lasercar commented:
> I rebased and tested the changes and the Ctrl-Alt-M keybind didn't seem to work anymore. I wonder if you can rebase and test if it's an issue with the PR or not.

https://github.com/user-attachments/assets/4344a1ca-9daf-4402-b0af-a2b0927c1fc1

Might have just been your fix the to null safety build error of this PR?

I've rebased to the latest develop so there should be no more weirdness

EliteMasterEric commented:
I'm on an unmodified `2d54076` and Ctrl-Alt-M does nothing, I didn't even rebase or make any changes.
